pkg/autodiscovery/providers/docker.go

```diff
@@ -120,16 +120,18 @@ func (d *DockerConfigProvider) IsUpToDate() (bool, error) {
 func parseDockerLabels(containers map[string]map[string]string) ([]integration.Config, error) {
 	var configs []integration.Config
 	for cID, labels := range containers {
-		c, err := extractTemplatesFromMap(docker.ContainerIDToEntityName(cID), labels, dockerADLabelPrefix)
-		switch {
-		case err != nil:
-			log.Errorf("Can't parse template for container %s: %s", cID, err)
-			continue
-		case len(c) == 0:
+		c, errors := extractTemplatesFromMap(docker.ContainerIDToEntityName(cID), labels, dockerADLabelPrefix)
+
+		if len(c) == 0 {
 			continue
-		default:
-			configs = append(configs, c...)
 		}
+		if len(errors) > 0 {
```

Comment:
nit: ranging on a `nil` slice is a no-op, so you can skip the length test and directly iterate